src/Microsoft.DotNet.Darc/src/DarcLib/LocalGitClient.cs

```diff
@@ -467,6 +454,70 @@ private static void CleanRepoAndSubmodules(LibGit2Sharp.Repository repo, ILogger
+        {
+            try
+            {
+                log.LogDebug($"Trying safe checkout of {repo.Info.WorkingDirectory} at {commit}");
+                LibGit2Sharp.Commands.Checkout(repo, commit, options);
+            }
+            catch (Exception e) when (e is LibGit2Sharp.InvalidSpecificationException || e is LibGit2Sharp.NameConflictException || e is LibGit2Sharp.LibGit2SharpException)
+            {
+                log.LogWarning($"Couldn't check out one or more files, possibly due to path length limitations ({e.ToString()}).  Attempting to checkout by individual files.");
+                SafeCheckoutByIndividualFiles(repo, commit, options, log);
+            }
+            catch
+            {
+                log.LogDebug($"Couldn't checkout {commit} as a commit after fetch.  Attempting to resolve as a treeish.");
+                string resolvedReference = ParseReference(repo, commit, log);
+                if (resolvedReference != null)
+                {
+                    log.LogDebug($"Resolved {commit} to {resolvedReference}, attempting to check out");
+                    try
+                    {
+                        log.LogDebug($"Trying checkout of {repo.Info.WorkingDirectory} at {resolvedReference}");
+                        LibGit2Sharp.Commands.Checkout(repo, resolvedReference, options);
+                    }
+                    catch (Exception e) when (e is LibGit2Sharp.InvalidSpecificationException || e is LibGit2Sharp.NameConflictException || e is LibGit2Sharp.LibGit2SharpException)
+                    {
+                        log.LogWarning($"Couldn't check out one or more files, possibly due to path length limitations ({e.ToString()}).  Attempting to checkout by individual files.");
+                        SafeCheckoutByIndividualFiles(repo, resolvedReference, options, log);
+                    }
+                }
+                else
+                {
+                    log.LogError($"Couldn't resolve {commit} as a commit or treeish.  Checkout of {repo.Info.WorkingDirectory} failed.");
+                    throw new ArgumentException($"Couldn't resolve {commit} as a commit or treeish.  Checkout of {repo.Info.WorkingDirectory} failed.");
+                }
+            }
+        }
+
+        private static void SafeCheckoutByIndividualFiles(LibGit2Sharp.Repository repo, string commit, LibGit2Sharp.CheckoutOptions options, ILogger log)
+        {
+            log.LogDebug($"Beginning individual file checkout for {repo.Info.WorkingDirectory} at {commit}");
+            SafeCheckoutTreeByIndividualFiles(repo, repo.Lookup(commit).Peel<LibGit2Sharp.Tree>(), "", commit, options, log);
+
+        }
+
+        private static void SafeCheckoutTreeByIndividualFiles(LibGit2Sharp.Repository repo, LibGit2Sharp.Tree tree, string treePath, string commit, LibGit2Sharp.CheckoutOptions options, ILogger log)
```

Comment:
>rivate static void SafeCheckoutTreeByIndividualFiles(LibGit2Sharp.Repository repo, LibGit2Sharp.Tree tree, string treePath, string commit, LibGit2Sharp.CheckoutOptions options, ILogger log) [](start = 9, length = 189)

Nit: split into shorter lines

Reply:
Changed as suggested.

---
In reply to: [285644577](https://github.com/dotnet/arcade-services/pull/422#discussion_r285644577) [](ancestors = 285644577)